### ISSUE TITLE
There was an error processing your request: An error occurred.

### ISSUE THREAD

**Issue by Soumyaranjan-17** [stackblitz-labs/bolt.diy#823]
### Describe the bug

I have **ollama** and also have **qwen2.5-coder:14b** model 
I have made a file name modelFile 

Q1: why it is requesting localhost:1234/v1/models

Ollama is running on its default port 

 ```
FROM qwen2.5-coder:14b
PARAMETER num_ctx 32768
```

After starting the server and a chat, it shows the error 
_There was an error processing your request: An error occurred._

### Link to the Bolt URL that caused the error

http://localhost:3000/chat/8

### Steps to reproduce

1. start the server 
2. start a chat

### Expected behavior

This will show a error like this 
![image](https://github.com/user-attachments/assets/d1227166-84e0-404a-8656-98a477b26432)

### Screen Recording / Screenshot

![image](https://github.com/user-attachments/assets/7c8a6d67-a43c-406a-9faf-538d60c3a6b2)
![image](https://github.com/user-attachments/assets/9f51de87-ad55-409f-86aa-75bbd46bcd3e)
![image](https://github.com/user-attachments/assets/0cd2f12d-2950-44f8-bb21-a57f91b91954)

Here I found the port 1234 

```
async function getLMStudioModels(_apiKeys?: Record<string, string>, settings?: IProviderSetting): Promise<ModelInfo[]> {
  try {
    const baseUrl = settings?.baseUrl || import.meta.env.LMSTUDIO_API_BASE_URL || 'http://localhost:1234';
    const response = await fetch(`${baseUrl}/v1/models`);
    const data = (await response.json()) as any;

    return data.data.map((model: any) => ({
      name: model.id,
      label: model.id,
      provider: 'LMStudio',
    }));
  } catch (e: any) {
    logStore.logError('Failed to get LMStudio models', e, { baseUrl: settings?.baseUrl });
    return [];
  }
}
```

if we can fix this ?

### Platform

- OS: [e.g. Window11]
- Browser: [Edge, Chrome]
- Version: [ Latest ]

**Comment by YusufMoh-FOM** [comment]
I have exactly the same Problem but I run bolt.diy on http://localhost:5173/. I also have ollama and qwen2.5-coder:14b. 

When starting the bolt.diy application, it immediately throws the following errors:
constants.ts:492 
GET http://localhost:1234/v1/models net::ERR_CONNECTION_REFUSED
getLMStudioModels @ constants.ts:492
(anonymous) @ constants.ts:528
initializeModelList @ constants.ts:528
(anonymous) @ BaseChat.tsx:158

This issue occurs repeatedly, specifically with the getLMStudioModels function. 

After attempting to write a prompt(Selected ollama => qwen2.5-coder:14b), the application fails with this additional error:
There was an error processing your request: An error occurred.
ERROR Chat Request failed logger.ts:84


I have ollama installed and its confirmed that it works and running.

I also use Windows 10 and Chrome as Browser

**Comment by Soumyaranjan-17** [comment]
Hii
Can you share some details 

1. Which model you are running 🦙 
2. If you are using ollama 
> Make sure to start the ollama server 
> Don't use http://localhost:11434
> ### Instead use http://127.0.0.1:11434
> Because IPV6 will convert localhost:11434 to ::1:11434 & that will cause a error

If you have some error 
Send the screenshot of
1. .env (where you have the OLLAMA API BASE URL)
2. Terminal console error 
3. Browser terminal error 
4. Setting page of bolt (provider)

**Comment by Yusuf-Moh** [comment]
Hey, 
I have installed Ollama and set up both the qwen2.5-coder:14b and qwen2.5-coder:32b models. Here's what I have done so far:
I created a folder named modelfile and a file inside it named modelfile.

First I ran it with this content:
FROM qwen2.5-coder:32b
PARAMETER num_ctx 32768

The second time I ran it with: 
FROM qwen2.5-coder:14b
PARAMETER num_ctx 32768

So I can use both Models. (I am currently using the 14b model.)

---
After that I checked my connection to Ollama via http://127.0.0.1:11434/
Ollama is running on its default port:
![image](https://github.com/user-attachments/assets/80fbf85c-31d9-4983-8a56-1aa93e895fa1)

My .env file looks like this for OLLAMA_API_BASE_URL
![image](https://github.com/user-attachments/assets/42885d8d-5454-4a4b-b60c-02787f99603a)

Here is my Setting page of bolt:
![image](https://github.com/user-attachments/assets/a560ed46-eb6f-4cb3-99aa-88bcba6150d4)


---
If I visit the Bolt Website at http://localhost:5173/ the **Browser console give me these two errors (its the same error):**
![image](https://github.com/user-attachments/assets/7b933dc7-b6fd-43cb-918c-ccb4f4377d35)


---
My Docker Terminal Console looks like this if I start the bolt.diy:
![image](https://github.com/user-attachments/assets/a7e0414c-1a2f-438c-8a71-c6ef253a46f5)


---
After writing a Prompt I get the error: Error: An error occurred. (My Docker Terminal Console remains the same as before)
![image](https://github.com/user-attachments/assets/1d7051dc-041c-4002-b03f-0f4944523eda)

---
So basically the Problem is the function getLMStudioModels. But I also can't write a Prompt and get a Output from ollama
![image](https://github.com/user-attachments/assets/b754b421-d5c0-41cf-9fde-b2db31440127)

**Comment by iamEtornam** [comment]
I am having the same issue

**Comment by iamEtornam** [comment]
![Screenshot 2024-12-26 134048](https://github.com/user-attachments/assets/0173ed76-cbda-4ecd-a394-b4aedeef4119)
After i set `DEFAULT_NUM_CTX=2000` for some reason it worked but didn't open the text editor

**Comment by Lex-au** [comment]
Desperately needs support for v1-openai endpoints instead of forced chat completions.

**Comment by leex279** [comment]
Generic errors are most likely a problem with the API key you using or the provider itself. For Deepseek e.g. you have to pay credits to make the API work (its not free!).
For OpenAI you also need to pay at least 5 Euro I guess to get to Tier1 and usable API.

The **main** branch has extended logging for the terminal (new features). Check this out, try your prompt again and check the terminal for errors.
-----

@Lex-au open feature request, with description what exactly is needed, then we can discuss

-----
Closed for know: If still a problem, open a topic in community please to investigate deeper: https://thinktank.ottomator.ai/c/bolt-diy/bolt-diy-issues-and-troubleshooting/22
